### PR TITLE
[marshal] add types for v0.5

### PR DESCRIPTION
**Pull request by JamieMagee**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/marshal/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Type definitions for marshal 0.5
+// Project: https://github.com/clayzermk1/node-marshal
+// Definitions by: Jamie Magee <https://github.com/JamieMagee>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+export default class Marshal {
```
Comment (peterblazejewicz) [inline types/marshal/index.d.ts:8]:
```suggestion
declare class Marshal {
```
https://unpkg.com/browse/marshal@0.5.2/lib/marshal.js
old style module export with Function construtor, IMO

File: types/marshal/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    parsed?: unknown;
+
+    constructor(buffer: string, encoding?: BufferEncoding);
+
+    load(buffer: string, encoding?: BufferEncoding): this;
+
+    toString(encoding?: BufferEncoding): string;
+    toJSON(): unknown;
+}
```
Comment (peterblazejewicz) [inline types/marshal/index.d.ts:17]:
```suggestion
}

export = Marshal;
```

File: types/marshal/marshal-tests.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import Marshal from 'marshal';
```
Comment (peterblazejewicz) [inline types/marshal/marshal-tests.ts:1]:
```suggestion
import Marshal = require('marshal');
```
https://github.com/clayzermk1/node-marshal#basic-use

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@JamieMagee thx!

**Comment by JamieMagee** [comment]
Ready to merge

**Pull request merged**